### PR TITLE
Added ready promise on init

### PR DESCRIPTION
**Pull request by lemieux**
@alavers @jpjoyal @juliangarritano

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,10 +59,10 @@ SupportKit.init({appToken: 'your_app_token'});
 ### Individual functions
 
 #### init(options)
-Initializes the SupportKit widget in the web page using the specified options
+Initializes the SupportKit widget in the web page using the specified options. It returns a promise that will resolve when the widget is ready if you need to do different things at different times.
```
Comment (juliangarritano) [inline README.md]: "If you need to do different things at different times"
Wut?

Comment (lemieux) [inline README.md]: Better phrasing? It allows you to keep a reference where you can always be sure the widget is init-ed

Comment (juliangarritano) [inline README.md]: "Initializes the SupportKit widget in the web page using the specified options. It returns a promise that will resolve when the widget is ready. Keeping a reference to that promise allows you to run code when you know init has completed."

I still don't like it.

**Comment by williamsjj** [comment]
If code adds an init promise later after init has long since completed, will the promise fire immediately or just queue waiting for some logout/init resequence that may never happen?

**Comment by juliangarritano** [comment]
As long as you keep a reference to the initial init call, any subsequent promises will fire immediately given it has been resolved.

**Comment by williamsjj** [comment]
@juliangarritano Awesome. Thanks for clarifying.

**Comment by lemieux** [comment]
The promise resolves at the same time `ready` is triggered. The main difference from using `ready` is that ready is a one time thing  while the promise can be `promise.then(...)` again and again.

**Comment by alavers** [comment]
:+1:

**Pull request merged**